File: core/crates/primitives/src/testkit/asset_metadata_mock.rs
use crate::AssetMetaData;

impl AssetMetaData {
    pub fn mock() -> Self {
        Self {
            is_enabled: true,
            is_balance_enabled: true,
            is_buy_enabled: false,
            is_sell_enabled: false,
            is_swap_enabled: false,
            is_stake_enabled: false,
            is_earn_enabled: false,
            is_pinned: false,
            is_active: true,
            staking_apr: None,
            earn_apr: None,
            rank_score: 0,
        }
    }
}

File: android/gemcore/src/testFixtures/kotlin/com/gemwallet/android/testkit/AssetMetaDataMock.kt
package com.gemwallet.android.testkit

import com.wallet.core.primitives.AssetMetaData

fun mockAssetMetaData(
    isSellEnabled: Boolean = false,
    isStakeEnabled: Boolean = false,
    isPinned: Boolean = false,
    stakingApr: Double? = null,
) = AssetMetaData(
    isEnabled = true,
    isBalanceEnabled = true,
    isBuyEnabled = false,
    isSellEnabled = isSellEnabled,
    isSwapEnabled = false,
    isStakeEnabled = isStakeEnabled,
    isEarnEnabled = false,
    isPinned = isPinned,
    isActive = true,
    rankScore = 1,
    stakingApr = stakingApr,
    earnApr = null,
)
